chat-plugins/mafia.js

```diff
@@ -354,6 +354,7 @@ class MafiaTracker extends Rooms.RoomGame {
 		if (!player && this.dead[user.userid] && this.dead[user.userid].restless) player = this.dead[user.userid];
 		if (!(target in this.players) && target !== 'nolynch') return false;
 		if (player.lynching || (target === player.userid && !this.selfEnabled)) return false;
+		if (target === player.userid && !this.lynches[target] && this.lynches[target].count < this.getHammer() - 1 && this.selfEnabled === 'hammer') return false;
```

Comment:
"if theyre lynching themselves and they dont have a lynch object and their (nonexistent) lynch object's count is one off the hammer"